observ/store.py

```diff
@@ -24,8 +24,11 @@ def inner(self, *args, **kwargs):
             current = to_raw(self.state)
             fn(self, *args, **kwargs)
             ops, reverse_ops = patchdiff.diff(current, self.state)
-            self._past.append((ops, reverse_ops))
-            self._future.clear()
+            # If ops and reverse_ops are empty, that means
+            # that there are no actual changes to record
+            if ops or reverse_ops:
+                self._past.append((ops, reverse_ops))
+                self._future.clear()
```

Comment:
Personally I would consider it a bug and expect an exception if I call a mutation that doesn't change anything. :/ But I can see a use case for preferring it to silently do nothing.

I propose adding a `strict=True` parameter to the `Store` constructor to toggle this behaviour. I think users should consciously opt into silent ignores of an edge case situation.

Reply:
No exception?

Reply:
Ah, right: forgot!